webapp/db/db.py

```diff
@@ -32,19 +33,38 @@ def create_challenge_session(client_name: str) -> models.ChallengeInitiateRespon
         return models.ChallengeInitiateResponse(
             id_secret=client_challenge.id_secret, challenge_secret=client_challenge.challenge_secret)
         
-def complete_challenge_session(client_name: str, challenge_secret: str):
-    """ Resolve a challenge session in the database """
+
+def _get_pending_auth_session(session: Session, client_name: str, challenge_secret: str) -> DbClientAuthSession:
+    client = session.scalar(select(DbClient).where(DbClient.name == client_name).where(DbClient.valid == True))
+    if client is None:
+        raise HTTPException(404, "Given client name is invalid")
```

Comment:
```suggestion
        raise HTTPException(401, "Given client name is invalid")
```
[401 unauthorized](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401) is probably more appropriate here.